elastic/kibana
# [Stack Monitoring] Verify remote cluster client role when CCS is enabled

Fixes #129546

### Summary
When visiting the Stack Monitoring app, the overview page first tries to load the clusters but if any permissions are missing you get a 403 on that request which causes the app to redirect you to the access denied page.

On this page we now display a helper text stating that you might be missing the `remote_cluster_client` role in your "local" cluster.

<img width="1391" alt="Screenshot 2022-09-14 at 16 32 08" src="https://user-images.githubusercontent.com/2564140/190186071-2f740dcd-cacf-4f04-b8f5-4e363de03583.png">

The access denied page also tries to check with the API what might be missing and if it's been resolved every 5 seconds.
Before this PR, that API check said everything was fine so it would try to return to the overview page again only to bounce back to the access denied page.

In this PR I made the API report a 403 if the remote cluster client role is missing so there is a bit less network traffic until it's resolved. 

<img width="1363" alt="Screenshot 2022-09-14 at 16 32 49" src="https://user-images.githubusercontent.com/2564140/190186836-be246eba-d53d-460d-bdf2-dfc70bb220f1.png">

I've also made this a specific error in the Kibana error log to make it easier to catch.

<img width="1332" alt="Screenshot 2022-09-14 at 16 33 31" src="https://user-images.githubusercontent.com/2564140/190186922-e599eed6-59dd-4a1d-abc9-eb9247eba870.png">


### How to test

Get the changes: `git fetch git@github.com:miltonhultgren/kibana.git 129546-sm-ccs-roles:129546-sm-ccs-roles && git switch 129546-sm-ccs-roles`

1. Locally setup two clusters to configure for CCR, following the steps in [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/dev_docs/how_to/running_components_from_source.md#multi-cluster-tests-for-ccrccs-or-listing) 
2. Edit the `elasticsearch.yml` of the cluster which the "local" and add this line: `node.roles: [master, data, data_content, data_hot, data_warm, data_cold, data_frozen, ingest, ml, transform]` (note that the role `remote_cluster_client` is missing from this list)
3. Ingest some Stack Monitoring data into the "remote" cluster
4. Visit the stack monitoring app and verify the above changes
